### PR TITLE
3.6 Node updateTransform apply recursive form

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: 

### Changelog

* Change Node::updateTransform into recursive form

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/core/scene-graph/Node.cpp [hunk 1]
```diff
@@ -530,68 +530,52 @@ void Node::setScaleInternal(float x, float y, float z, bool calledFromJS) {
...
+
+    if (parent) {
+        parent->updateWorldTransform();
+        if (dirtyBits & static_cast<uint32_t>(TransformBit::POSITION)) {
+            _worldPosition.transformMat4(_localPosition, parent->getWorldMatrix());
+            _worldMatrix.m[12] = _worldPosition.x;
+            _worldMatrix.m[13] = _worldPosition.y;
+            _worldMatrix.m[14] = _worldPosition.z;
         }
-        dirtyBits |= child->getDirtyFlag();
```
Comment (dumganhar) [inline native/cocos/core/scene-graph/Node.cpp]: Missing the logic of this line

Comment (PatriceJiang) [reply native/cocos/core/scene-graph/Node.cpp]: fixed

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

**Pull request closed without merging**